Makefile

```diff
@@ -98,4 +98,4 @@ clean:
 
 .PHONY: e2e
 e2e:
-	$(GO) run github.com/onsi/ginkgo/ginkgo --v --randomizeAllSpecs --randomizeSuites --race $(TAGS) ./test/e2e
+	$(GO) run github.com/onsi/ginkgo/ginkgo --v --randomizeAllSpecs --randomizeSuites --race $(if $(TEST),-focus '$(TEST)') $(TAGS) ./test/e2e -- $(if $(SKIPTLS),-skip-tls true) 
```

Comment:
Looks at some of the other review comments, and the e2e_tests.md in this PR, it looks like we're getting to the point where we might want to use a helper bash script that this `make e2e` target will call, that will do the heavy-lifting for us, like checking if the `KUBECONFIG` or enforcing the version of kind, etc. We can evaluate that outside of this PR though.

Reply:
Please see the latest changes that I pushed. I've implemented Joe's request for bringing forward the "provisioning" code that was in OLM.

Reply:
Oh man, that's awesome - I was under the impression we were only going to tackle documenting running e2e tests locally as that wasn't immediately clear to me as well when I started diving into the registry codebase recently, but attempting to automate that as well is super nice.